sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.getStudentApplicationSummary.e2e-spec.ts

```diff
@@ -265,6 +266,58 @@ describe("StudentInstitutionsController(e2e)-getStudentApplicationSummary", () =
     },
   );
 
+  it("Should throw forbidden error when the institution type is not BC Public.", async () => {
+    // Arrange
+    // Student submitting an application to College C.
+    const { student, collegeCApplication } =
+      await saveStudentApplicationForCollegeC(appDataSource);
+
+    await authorizeUserTokenForLocation(
+      appDataSource,
+      InstitutionTokenTypes.CollegeCUser,
+      collegeCApplication.location,
+    );
+
+    // College C is not a BC Public institution.
+    const collegeCInstitutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeCUser,
+    );
+
+    const endpoint = `/institutions/student/${student.id}/application-summary?page=0&pageLimit=10&sortField=applicationNumber&sortOrder=${FieldSortOrder.DESC}`;
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(collegeCInstitutionUserToken, BEARER_AUTH_TYPE)
+      .expect({
+        statusCode: HttpStatus.FORBIDDEN,
+        message: "Forbidden resource",
```

Comment:
We can throw a specific `Forbidden resource` message to ensure that the validation is happening as we expected and stopping because of the right validation.
We do it already here, for instance: https://github.com/bcgov/SIMS/blob/main/sources/packages/backend/apps/api/src/auth/guards/authorized-parties.guard.ts#L45